src/pyodide/internal/loadPackage.ts

```diff
@@ -55,7 +65,7 @@ async function loadBundleFromR2(requirement: string): Promise<Reader> {
   const filename = getFilenameOfPackage(requirement);
   const cached = DiskCache.get(filename);
   if (cached) {
-    const decompressed = await decompressArrayBuffer(cached);
+    const decompressed = await decompressArrayBuffer(cached, filename);
```

Comment:
Let's leave `decompressArrayBuffer` alone and do:
```js
async function loadBundleFromR2(requirement: string): Promise<Reader> {
  // first check if the disk cache has what we want
  const filename = getFilenameOfPackage(requirement);
  let original = DiskCache.get(filename);
  if (!original) {
    const url = new URL(WORKERD_INDEX_URL + filename);
    const response = await fetch(url);

    original = await response.arrayBuffer();
    DiskCache.put(filename, original);
  }

  const uncompressed = (filename.endsWith('.tar.gz')) ? await decompressArrayBuffer(original) : original;
  const reader = new ArrayBufferReader(uncompressed);
  return reader;
}
```

Reply:
Done